crates/jxl-coding/src/error.rs

```diff
@@ -7,6 +7,8 @@ pub enum Error {
     InvalidAnsHistogram,
     InvalidAnsStream,
     InvalidPermutation,
+    TooManySplines(usize),
+    TooManySplinePoints(usize),
```

Comment:
I feel like `jxl_frame` is better choice for spline errors, as they occur in the frame decoding section. `jxl_coding` is for entropy coding.

Reply:
Thanks! Moved spline errors to `jxl_coding::Error`